youtube_dl/extractor/common.py

```diff
@@ -2830,7 +2830,10 @@ def _get_cookies(self, url):
         """ Return a compat_cookies.SimpleCookie with the cookies for the url """
         req = sanitized_Request(url)
         self._downloader.cookiejar.add_cookie_header(req)
-        return compat_cookies.SimpleCookie(req.get_header('Cookie'))
+        cookie_header = req.get_header('Cookie')
+        if sys.version_info[0] == 2:
+            cookie_header = str(cookie_header)
```

Comment:
1. This won't work for UTF-8.
2. This will produce `"None"` string if there is no cookies, i.e. `cookie_header` is `None` that is completely wrong.